### PR TITLE
API Review for Http.Connections

### PR DESCRIPTION
**Pull request by anurse**
* Microsoft.AspNetCore.Http.Connections
  * Remove string overloads of MapConnectionHandler, MapConnections on ConnectionsRouteBuilder
  * Pubternalize HttpConnectionContext
  * Unnest HttpConnectionContext.ConnectionStatus (and Pubternalize)
  * Pubternalize HttpConnectionDispatcher
  * Pubternalize HttpConnectionManager
  * Rename `HttpConnectionOptions` (ideas: `HttpConnectionDispatcherOptions`)

### CODE REVIEW

File: src/Microsoft.AspNetCore.Http.Connections/WebSocketOptions.cs [hunk 1]
```diff
@@ -2,13 +2,25 @@
...
 
 namespace Microsoft.AspNetCore.Http.Connections
 {
     public class WebSocketOptions
     {
         public TimeSpan CloseTimeout { get; set; } = TimeSpan.FromSeconds(5);
 
-        public string SubProtocol { get; set; }
+        /// <summary>
+        /// Set this to a delegate that will be called when a new WebSocket is established to select the value
```
Comment (JamesNK) [inline src/Microsoft.AspNetCore.Http.Connections/WebSocketOptions.cs]: Gets or sets a delegate that...

File: src/Microsoft.AspNetCore.Http.Connections/HttpConnectionDispatcherOptions.cs [hunk 2]
```diff
@@ -7,13 +7,13 @@
 
 namespace Microsoft.AspNetCore.Http.Connections
 {
-    public class HttpConnectionOptions
+    public class HttpConnectionDispatcherOptions
```
Comment (JamesNK) [inline src/Microsoft.AspNetCore.Http.Connections/HttpConnectionDispatcherOptions.cs]: 👍

File: src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionStatus.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionStatus.cs]: nit: remove newline

Comment (anurse) [reply src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionStatus.cs]: remove and sort seems to always leave these sitting around :(

File: src/Microsoft.AspNetCore.Http.Connections/WebSocketOptions.cs (hunk 1 above)
Comment (anurse) [reply src/Microsoft.AspNetCore.Http.Connections/WebSocketOptions.cs]: Derp. Good call

**Pull request merged**